### ISSUE TITLE
do notation

### ISSUE THREAD

**Issue by JasonShin** [JasonShin/functional-programming-jargon.rs#22]
Based on the Monad implementation, we should be able to explain Haskell's `do notation`, potentially creating an emulation of it as well.

Take a look at an existing `do` notation crate https://github.com/TeXitoi/rust-mdo for an inspiration.

References:
- https://en.wikibooks.org/wiki/Haskell/do_notation
- https://docs.scala-lang.org/tour/for-comprehensions.html
- http://learnyouahaskell.com/a-fistful-of-monads

**Comment by Boscop** [comment]
Yes, please impl Monad for more types and provide a do-notation like rust-mdo :)
I'm really looking forward to this..
